code/game/atoms_movable.dm

```diff
@@ -93,17 +93,13 @@
 /atom/movable/Crossed(atom/movable/AM)
 	return
 
-/atom/movable/Bump(var/atom/A as mob|obj|turf|area, yes)
-	if(src.throwing)
-		src.throw_impact(A)
-		src.throwing = 0
-
-	if ((A && yes))
-		A.last_bumped = world.time
+/atom/movable/Bump(var/atom/A as mob|obj|turf|area, yes) //the "yes" arg is to differentiate our Bump proc from byond's, without it every Bump() call would become a double Bump().
+	if((A && yes))
+		if(throwing)
+			throwing = 0
+			throw_impact(A)
+			. = 1
 		A.Bumped(src)
-	return
-	..()
-	return

```

Comment:
rip in peace

On a more serious note, this shouldn't have the as mob|obj|turf|area bit.